syntax = 'proto3';

package com.ssinchenko.tsumugi.proto;

option java_multiple_files = true;
option java_package = "com.ssinchenko.tsumugi.proto";
option java_generate_equals_and_hash = true;
option optimize_for=SPEED;
option go_package = "tsumugi/proto";

message Repository {
  oneof repository {
    FileSystemRepository file_system = 1;
    SparkTableRepository spark_table = 2;
  }
}

message FileSystemRepository {
  string path = 1;
}

message SparkTableRepository {
  string table_name = 1;
}

message ResultKey {
  int64 dataset_date = 1;
  map<string, string> tags = 2;
}
